src/Pos/Block/Network/Logic.hs

```diff
@@ -270,16 +270,19 @@ requestHeaders
     -> ConversationActions MsgGetHeaders (LimitedLength s (MsgHeaders ssc)) m
     -> m ()
 requestHeaders cont mgh nodeId _ conv = do
-    logDebug $ sformat ("requestHeaders: withConnection: sending "%build) mgh
+    logDebug $ sformat ("requestHeaders: sending "%build) mgh
     send conv mgh
     mHeaders <- recvLimited conv
     inRecovery <- recoveryInProgress
     logDebug $ sformat ("requestHeaders: inRecovery = "%shown) inRecovery
     flip (maybe onNothing) mHeaders $ \(MsgHeaders headers) -> do
         logDebug $ sformat
-            ("requestHeaders: withConnection: received "%listJson%
-             " from nodeId "%build%" of total size "%builder)
-            (map headerHash headers) nodeId (unitBuilder $ biSize headers)
+            ("requestHeaders: received "%int%" headers of total size "%builder%
+             " from nodeId "%build%": "%listJson)
+            (headers ^. _Wrapped . to NE.length)
```

Comment:
Minor: I think you could've used `Foldable.length` here and avoid lenses/unwrapping.